subliminal/providers/legendastv.py

```diff
@@ -359,8 +375,16 @@ def query(self, language, title, season=None, episode=None, year=None):
                     if not a.pack and 'episode' in guess and guess['episode'] != episode:
                         continue
 
+                # calculates a good expiration time
+                if a.timestamp:
+                    delta = max(0, (datetime.datetime.now() - a.timestamp).total_seconds())
+                    # recently uploaded archives have short expiration time
+                    expiration_time = min(RELEASES_EXPIRATION_TIME, delta / 8)
+                else:
+                    expiration_time = RELEASES_EXPIRATION_TIME
+
                 # attempt to get the releases from the cache
-                releases = region.get(releases_key.format(archive_id=a.id))
+                releases = region.get(releases_key.format(archive_id=a.id), expiration_time=expiration_time)
```

Comment:
Well if it's not in the cache it will be cached and if we want to get it another time it will get it from cache.
What's the issue with getting the issue from the cache if we know it hasn't changed?